modules/KalturaSupport/resources/mw.ClosedCaptions.js

```diff
@@ -664,12 +664,20 @@
 		},
 		selectDefaultIosTrack: function (defaultLangKey) {
 			var _this = this;
-			this.once( 'playing', function (){
-				setTimeout(function () {
-					_this.log('selectDefaultIosTrack: ' + defaultLangKey);
-					_this.embedPlayer.selectDefaultCaption(defaultLangKey);
-				}, 500);
-			});
+			if (_this.embedPlayer.isPlaying()){
+				_this.selectTextTrack(defaultLangKey);
+			} else {
+				_this.once( 'playing', function (){
+					_this.selectTextTrack(defaultLangKey);
+				});
+			}
+		},
+		selectTextTrack: function(defaultLangKey) {
+			var _this = this;
+			setTimeout(function () {
```

Comment:
@ErezBouskila better keep this timeout id in a var and if change media is done while it is still pending then clear it

Reply:
@oren this is a timeout (one of million in V2...) not an interval. why to keep it on this?

Reply:
LGTM